### PR TITLE
Masked loss funcs

### PR DESCRIPTION
**Pull request by dsm-72**
Adds masked losses for use with `torch.nn.rnn` that lets users calculate some losses with a padded sequence. Example of usage (before this PR) is in the [GitHub Gist][Github Gist] shown in [PackedSequences on MPS accelerator][PackedSequences on MPS accelerator].

[clarification for initialization strategies]: https://github.com/pytorch/pytorch/issues/102730
[Github Gist]: https://gist.github.com/dsm-72/1cea0601145a8b92155d8d08c90bf998
[PackedSequences on MPS accelerator]: https://github.com/pytorch/pytorch/issues/102911

### CODE REVIEW

**Comment by albanD** [comment]
Hi,

Thanks for proposing this. It would be great to open an issue so that we can discuss adding such a feature with more context and discuss the design there.

**Comment by dsm-72** [comment]
@albanD please see this [issue](https://github.com/pytorch/pytorch/issues/102971) then!
